src/Simplify.elm

```diff
@@ -7706,6 +7667,69 @@ introducesVariableOrUsesTypeConstructor context nodesToLookAt =
                     introducesVariableOrUsesTypeConstructor context remaining
 
 
+appliedLambdaChecks : { lambdaRange : Range, lambda : Expression.Lambda, firstArgument : Node Expression } -> List (Error {})
+appliedLambdaChecks { lambdaRange, lambda, firstArgument } =
+    case lambda.args of
+        (Node unitRange Pattern.UnitPattern) :: otherPatterns ->
+            [ Rule.errorWithFix
+                { message = "Unnecessary unit argument"
+                , details =
+                    [ "This function is expecting a unit, but also passing it directly."
+                    , "Maybe this was made in attempt to make the computation lazy, but in practice the function will be evaluated eagerly."
+                    ]
+                }
+                unitRange
+                (case otherPatterns of
+                    [] ->
+                        [ Fix.removeRange { start = lambdaRange.start, end = (Node.range lambda.expression).start }
+                        , Fix.removeRange (Node.range firstArgument)
+                        ]
+
+                    secondPattern :: _ ->
+                        [ Fix.removeRange { start = unitRange.start, end = (Node.range secondPattern).start }
+                        , Fix.removeRange (Node.range firstArgument)
+                        ]
+                )
+            ]
+
+        (Node allRange Pattern.AllPattern) :: otherPatterns ->
+            [ Rule.errorWithFix
+                { message = "Unnecessary wildcard argument argument"
+                , details =
+                    [ "This function is being passed an argument that is directly ignored."
+                    , "Maybe this was made in attempt to make the computation lazy, but in practice the function will be evaluated eagerly."
+                    ]
+                }
+                allRange
+                (case otherPatterns of
+                    [] ->
+                        [ Fix.removeRange { start = lambdaRange.start, end = (Node.range lambda.expression).start }
+                        , Fix.removeRange (Node.range firstArgument)
+                        ]
+
+                    secondPattern :: _ ->
+                        [ Fix.removeRange { start = allRange.start, end = (Node.range secondPattern).start }
+                        , Fix.removeRange (Node.range firstArgument)
+                        ]
+                )
+            ]
+
+        _ ->
+            [ Rule.error
+                { message = "Found immediately invoked anonymous function"
```

Comment:
You do this in a few error messages: "Found ...". why not just say what you found? Like
> Anonymous function is immediately invoked

Reply:
Yeah, I keep getting confused about what I should say especially for `Simplify` (there are so many errors that there's not even a good template I feel like I could follow :sweat_smile: )

Your suggestion sounds really good :+1: